addon/-private/system/model/model.js

```diff
@@ -808,11 +808,8 @@ const Model = Ember.Object.extend(Ember.Evented, {
   },
 
   willDestroy() {
-    //TODO Move!
     this._super(...arguments);
-    this._internalModel.clearRelationships();
-    this._internalModel.recordObjectWillDestroy();
-    //TODO should we set internalModel to null here?
+    this._internalModel.resetRecord();
```

Comment:
Seems odd to call this from the record into the internalModel. Can we move it somewhere else?